java/client/src/org/openqa/selenium/chrome/ChromeDriver.java

```diff
@@ -179,6 +180,10 @@ public ChromeDriver(ChromeDriverService service, ChromeOptions options) {
   @Deprecated
   public ChromeDriver(ChromeDriverService service, Capabilities capabilities) {
     super(new ChromeDriverCommandExecutor(service), capabilities);
+    if (capabilities.getCapability("enableDownloading")!=null){
+      sendCommandForDownloadChromeHeadLess(
+          (String) capabilities.getCapability("enableDownloading"));
```

Comment:
You may get a `ClassCastException` here. There are two ways to avoid this: either make the check above be `capabilities.getCapability("enableDownloading") instanceof String`, or use `String.valueOf` here.